File: solve/0408_실패율/postforty/실패율.py
def solution(N, stages):
    answer = []

    count = len(stages)

    stage_dict = {}

    for i in range(1, N+1):
        stage_dict[i] = 0
    
    for i in stages:
        try:
           stage_dict[i] += 1
        except:
            pass

    for i in range(1, N+1):
        temp = stage_dict[i]
        try:
            stage_dict[i] = stage_dict[i] / count
        except:
            stage_dict[i] = 0
        finally:
            count -= temp

    dic_to_list = stage_dict.items()
    result = sorted(dic_to_list, key=lambda x: x[1], reverse=True)

    answer = [i[0] for i in result]

    return answer

print(solution(4, [1, 1, 1, 1, 1]))

File: solve/0408_실패율/postforty/실패율.js
function solution(N, stages) {
  var answer = [];

  let count = stages.length;

  const stageObj = {};
  for (let i = 1; i <= N; i++) {
    stageObj[i] = 0;
  }

  for (let i of stages) {
    stageObj[String(i)] += 1;
  }

  tempArr = [];

  let stageArr = Object.entries(stageObj);

  for (let i = 0; i < N; i++) {
    tempArr.push(
      isNaN(stageArr[i][1] / count)
        ? [stageArr[i][0], 0]
        : [stageArr[i][0], stageArr[i][1] / count]
    );
    count -= stageArr[i][1];
  }

  tempArr.sort((a, b) => b[1] - a[1]);

  answer = tempArr.map((el) => Number(el[0]));

  return answer;
}

console.log(solution(4, [5, 5, 5, 5, 5]));
